include/boost/fusion/container/list/detail/list_to_cons.hpp

```diff
@@ -13,6 +13,61 @@
 ///////////////////////////////////////////////////////////////////////////////
 // Without variadics, we will use the PP version
 ///////////////////////////////////////////////////////////////////////////////
+#if !defined(BOOST_FUSION_HAS_VARIADIC_LIST)
 # include <boost/fusion/container/list/detail/cpp03/list_to_cons.hpp>
+#else
 
+///////////////////////////////////////////////////////////////////////////////
+// C++11 interface
+///////////////////////////////////////////////////////////////////////////////
+#include <boost/fusion/container/list/cons.hpp>
+
+namespace boost { namespace fusion { namespace detail
+{
+    template <typename ...T>
+    struct list_to_cons;
+
+    template <>
+    struct list_to_cons<>
+    {
+        typedef nil_ type;
+    };
+
+    template <typename Head>
+    struct list_to_cons<Head>
+    {
+        typedef Head head_type;
+        typedef list_to_cons<> tail_list_to_cons;
+        typedef typename tail_list_to_cons::type tail_type;
+
+        typedef cons<head_type, tail_type> type;
+
+        BOOST_FUSION_GPU_ENABLED
+        static type
+        call(typename detail::call_param<Head>::type _h)
+        {
+            return type(_h);
+        }
+    };
```

Comment:
I think it should be possible to drop this specialization by simply giving `list_to_cons<>` a `call` static member function. Have you considered it?